tests/functional/p/possible_f_string_as_string.py

```diff
@@ -0,0 +1,12 @@
+# pylint: disable=missing-module-docstring, invalid-name
+var = "string"
+var_two = "extra string"
+
+x = f"This is a {var} which should be a f-string"
+x = "This is a {var} used twice, see {var}"
+x = "This is a {var} which should be a f-string"  # [possible-f-string-as-string]
+x = "This is a {var} and {var_two} which should be a f-string"  # [possible-f-string-as-string]
+x1, x2, x3 = (1, 2, "This is a {var} which should be a f-string")  # [possible-f-string-as-string]
+
+y = "This is a {var} used for formatting later"
+z = y.format(var="string")
```

Comment:
I think we need some more tests here. We usually have a tons of false positive when adding a new checker because there is a lot of use case and some coders do really unexpected things when there is so much users. In the worst case it can be crashes. I guess we could start by adding actual pylint f-string and remove the f here. There can be expression inside f-strings too for example ``"This is an example with a list: {''.join(examples) + 'well...' }" that should not crash.``

Reply:
I wonder if there is a better way to do this. Similar to the problem identified in the OP with calculations not involving any variables. Those should ideally also be added.

Can we use `eval`? And just check and see if the `{}` and its contents change when they are evaluated as part of an f-string?

Reply:
I'd rather we not use ``eval``, pylint is launched on a lot of continuous integration and this would create an attack vector to execute random code contained in a string with containing ``{}``. Or did I misunderstand ?

Reply:
No, you understood and this is what I expected. I hoped there would perhaps be a 'utils' function that allow us to check for this.
Any other good ideas how to do this? Especially the calculations without variables seems difficult to do without some sort of evaluation/comparison between both string types.

Reply:
Could you elaborate on the "calculations without variables" part ? 😄

Reply:
'f"This example {1 + 1}"' for example...

Reply:
Well, I suggested hastily something that is probably not very reasonable. We'll always have false positive if we check for more than local variable (think of someone generating python code inside a dict). Maybe the sane thing to do it to limit false positive even if we have a lot of false negative for this check and warn only if a local var is detected. But maybe we could create an optional checker that is really aggressive and use ``ast.parse`` in order to check if something look like valid python code then warn for every string with parseable python code inside ``{}`` ?

Reply:
I think that is the best way forward. However, does that aggressive check need to be optional? I don't have as much experience with python code as you do, but does it happen often that valid python code is intentionally put inside `{}` within a normal string?

Reply:
> does it happen often that valid python code is intentionally put inside {} within a normal string?

You're right, probably not... pylint is notorious for its false positives and I wanted to be cautious 😄 

I'm even struggling to come up with an example. Here's a silly example that work because I think ``first_key: "int"`` is valid python:

```python
generated_python = """
first_key = "version"
DYNAMIC_DICT = {
first_key: "int"
}
"""
```

Reply:
There is also an example in the original issue. However the valid python also needs to return or represent something. Your example is not a valid f-string. I'm not sure if astroid allows us to check for this?

Reply:
You're right it also needs to return something... Maybe there would not be that much false positive after all. astroid can parse python too and infer the value of things, but there is not check for "this string is a valid python code" or "this string is a valid f-string as far as I know".

Reply:
I don't know the best way forward here. After googling a bit, f-strings seem to be fairly complicated as well as tricky to parse. 
Adding a checker for local variables in normal strings containing "{}" could be a partial solution, but I feel whatever we end up writing for that checker will be incompatible with a full-fledged `possible-f-string` checker. Therefore, I don't know if that warrants the time-investment.

Reply:
I think a problem here is that `ast.parse` is not able to recognise `bar` as a variable from the local and global scope, and I don't think we are able to tell it do so.
I have played around with letting `ast.parse` parse all lines up until the line with the string, changing the string to an f-string and comparing the difference. However, if the string is at the bottom of a 300+ lines file this would have major performance impact (I think). 
Other than that, I had no real ideas of how to supply local and global variables included in the f-string to `ast.parse`)

Reply:
Ha yes, you're right. We have to parse it in the context of our function. We can probably try to infer ``'\n'.join(bar)`` as if it's code inside the function at the position of the string using astroid (or ast.parse for that matter but when analyzing the core around the string we're supposed to have a parsed context in astroid already, so it should cost less). This is clearly non trivial and we should consider performance implication as you said. How about we focus on the easy case with only local_variable ?

Reply:
I think the problem with fixing the easy case now is that it is almost certainly going to be incompatible with a full implementation of a f-string checker. For me, this is a reason to consider whether the time investment needed to make this work (although most of the work is already present in this PR) is warranted.

Reply:
Clearly. In term of maintainer time as well, we'll get swamped with false negatives report as soon as we add the check in a live version. User will expect it to work properly on all use case (as they should). Might as well do it properly from the beginning and avoid the pain.  I'll add the work in progress label.